Projects/b_u585i_iot02a_ntz/Src/ota_pal/ota_pal_stm32u5_ntz.c

```diff
@@ -375,390 +591,565 @@ static HAL_StatusTypeDef prvWriteToFlash( uint32_t destination,
+                        xOtaStatus = OtaPalSuccess;
+                        pxContext->xPalState = OTA_PAL_ACCEPTED;
+                    }
+
+                    /* Delete context from flash */
+                    ( void ) prvDeleteImageNvContext();
                 }
 
                 break;
 
-            case OtaImageStateAccepted:
-
-                if( ( pContext->state == OTA_PAL_IMAGE_STATE_PENDING_COMMIT ) &&
-                    ( pContext->ulBank == activeBank ) )
-                {
-                    /** New image bank is booted successfully and it have passed self test. Make it as accepted
-                     * by removing the image context from flash and setting image state to valid. */
-
-                    prvRemoveImageContextFromFlash();
+            case OtaImageStateRejected:
 
-                    ( void ) prvEraseBank( passiveBank );
-                    pContext->state = OTA_PAL_IMAGE_STATE_COMMITTED;
-                    otaStatus = OtaPalSuccess;
-                }
-                else
+                switch( pxContext->xPalState )
                 {
-                    otaStatus = OtaPalCommitFailed;
+                    case OTA_PAL_PENDING_SELF_TEST:
+                    case OTA_PAL_NEW_IMAGE_BOOTED:
+                    case OTA_PAL_NEW_IMAGE_WDT_RESET:
+                    case OTA_PAL_PENDING_ACTIVATION:
+                    case OTA_PAL_FILE_OPEN:
+
+                        if( ( prvSelectBank( ulGetOtherBank( pxContext->ulTargetBank ) ) == pdTRUE ) &&
+                            ( prvEraseBank( pxContext->ulTargetBank ) == pdTRUE ) &&
+                            ( prvDeleteImageNvContext() == pdTRUE ) )
+                        {
+                            xOtaStatus = OtaPalSuccess;
+                            pxContext->xPalState = OTA_PAL_REJECTED;
+                        }
+
+                        break;
+
+                    case OTA_PAL_REJECTED:
+                        xOtaStatus = OtaPalSuccess;
+                        break;
+
+                    default:
+                        break;
                 }
 
                 break;
 
-            case OtaImageStateRejected:
+            case OtaImageStateTesting:
 
-                /* Make sure the image is not committed already. A committed image cannot be aborted or rejected. */
-                if( pContext->state != OTA_PAL_IMAGE_STATE_COMMITTED )
-                {
-                    /*
-                     *  Remove the persisted state of image from flash.
-                     */
-                    prvRemoveImageContextFromFlash();
-                    pContext->state = OTA_PAL_IMAGE_STATE_INVALID;
-                    otaStatus = OtaPalSuccess;
-                }
-                else
+                switch( pxContext->xPalState )
                 {
-                    otaStatus = OtaPalRejectFailed;
+                    case OTA_PAL_PENDING_ACTIVATION:
+                    case OTA_PAL_PENDING_SELF_TEST:
```

Comment:
Why do you expect this block to be executed while in `OTA_PAL_PENDING_SELF_TEST` state